mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelProvider.kt

```diff
@@ -39,8 +39,13 @@ object MvRxViewModelProvider {
         forExistingViewModel: Boolean = false,
         initialStateFactory: MvRxStateFactory<VM, S> = RealMvRxStateFactory()
     ): VM {
-        val stateRestorer = viewModelContext
-            .savedStateRegistry
+        val savedStateRegistry = viewModelContext.savedStateRegistry
+
+        if (!savedStateRegistry.isRestored) {
+            error("You can only access a view model after onCreate of your component has been called.")
```

Comment:
maybe  change `component` to `Fragment`? could also specify that only `super.onCreate` needs to be called, viewmodels can be accessed in onCreate after the super call.

@BenSchwab how do you feel about  this new restriction? Previously it would work to access viewmodels after super.onAttach of the fragment, but I believe that doing so is currently broken, since the viewmodel accessed then can be replaced in the viewmodel store by another viewmodel that is created during the state restoration in onCreate.

So in practice  this restriction always existed and this enforces it, and any violations are currently bugs

Reply:
This could also apply for `Activities` as well? I update this to be more specific.